activerecord/lib/active_record/enum.rb

```diff
@@ -209,7 +256,7 @@ def _enum(name, values, prefix: nil, suffix: nil, scopes: true, instance_methods
 
         attribute(name, **options) do |subtype|
           subtype = subtype.subtype if EnumType === subtype
-          EnumType.new(name, enum_values, subtype)
+          EnumType.new(name, enum_values, subtype, raise_on_invalid_values: !validate)
```

Comment:
Is it expected for an `ArgumentError` to be raised when we explicitly include `validate: false` in the `enum` definition?

Reply:
Yes, it is

Reply:
If that is the case, why do we need the `validate` params, since we are raising the error for both cases?

Reply:
Yes, you don't need pass `validate: false`. Like in `belongs_to` you don't need to pass `optional: false` explicitly

Reply:
TLDR:

- when `:validate` is falsey (by default) — standard "old" behaviour with raising error
- when `:validate` is truthy (need explicit specify) — soft adding of validation message

Detailed:

`ArgumentError` raises only if the `validate` argument is falsy (`validate: nil` or `validate: false`) and only if you try to assign invalid value. It is current behaviour of `enum` before this PR. And PR suggests keep this behaviour as is. That's why these change don't break legacy code. And you don't need pass `validate: false` explicitly. It is redundant like with above example (`belongs_to :user, optional: false`  — it is unlikely somebody write)

If pass truthy `validate` argument, enum is validated with rails inclusion validation. If you try to assign invalid value, a validation error is added to the model instance (`ArgumentError` doesn't raise). But `ActiveRecord::RecordInvalid` can raise if use bang methods `save!`, `create!` etc.)

Please see examples in the PR description and tests. I think it will be more understandable

Reply:
That makes sense @mechnicov. Thanks :)